hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/validators/block/StateChangesValidator.java

```diff
@@ -246,16 +252,29 @@ public StateChangesValidator(
                 new ConfigProviderImpl().getConfiguration(),
                 networkInfo,
                 new NoOpMetrics());
-
         logger.info("Registered all Service and migrated state definitions to version {}", servicesVersion);
     }
 
     @Override
     public void validateBlocks(@NonNull final List<Block> blocks) {
         logger.info("Beginning validation of expected root hash {}", expectedRootHash);
+        var previousBlockHash = INITIAL_BLOCK_HASH;
+
         for (final var block : blocks) {
+            StateChanges lastStateChanges = null;
```

Comment:
As per the issue description we need to validate that `the final state change in the block is a BlockStreamInfo with previous block hash that matches the observed hash for block N-1 in the stream`.